pkg/image/resolve.go

```diff
@@ -119,20 +119,37 @@ func generateImages(chartNameAndVersion string, inputMap map[interface{}]interfa
 		return
 	}
 
-	// distinguish images by os
-	os := inputMap["os"]
-	switch os {
-	case "windows": // must have indicate `os: windows` if the image is using in Windows cluster
-		if osType != Windows {
-			return
-		}
-	default:
-		if osType != Linux {
-			return
+	imageName := fmt.Sprintf("%s:%v", repo, t)
+
+	// By default, images are added to the generic images list ("linux"). For Windows and multi-OS
+	// images to be considered, they must use a comma-delineated list (e.g. "os: windows",
+	// "os: windows,linux", and "os: linux,windows").
+	if inputMap["os"] != nil {
+		switch osValue := inputMap["os"].(type) {
+		case string:
+			foundMatchingOSType := false
```

Comment:
Why not just call `addSourceToImage` when OS is found and return.

Reply:
Good point. That's a much better idea.